### PR TITLE
Switch to Github Actions for CI

### PR DESCRIPTION
**Pull request by colby-swandale**
The RubyGems org has been slowly migrating rubygems/bundler away from Travis-CI to use Github for running CI. GH Actions gives us a ton more flexibility in terms performing automated tasks like CI, scheduling jobs or automatically responding to issues. I think it's worth we do the same for rubygems.org.

I've have kept most of the behavior from Travis-CI the same, except for testing with `ruby-head` as It's not really possible to do it with this Github Actions setup. And the build in Travis-CI also seems to have been failing for quite a while now, so i think it's safe to drop it.

### CODE REVIEW

**Comment by colby-swandale** [comment]
Just note that the RubyGems 2.6.10 will fail as we have not yet configured Github with the keys for quay.io

**Comment by dwradcliffe** [comment]
Ruby-head was a request from the ruby core team, to assist in their testing. I wonder if there's another way we can provide this for them, perhaps outside the normal testing workflow?

**Comment by dwradcliffe** [comment]
Could we do the image build+push in a separate (parallel) step? Also, splitting up the different tests into steps so it's clear what failed from the list of checks would be neat.

**Comment by deivid-rodriguez** [comment]
For testing against ruby-head in Github Actions, we've used two approaches at https://github.com/rubygems/rubygems:

* Installing it through rvm just like TravisCI does.
* Use https://github.com/ruby/ruby-docker-images.

**Comment by colby-swandale** [comment]
Added testing for ruby-head but it will fail like it has been in Travis-CI. I'll take a look and get that fixed before merging this.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  BUNDLE_JOBS: 3
+  BUNDLE_RETRY: 3
+jobs:
+  build:
+    name: Test Application
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        rubygems: [2.6.10, latest]
+    services:
```
Comment (simi) [inline .github/workflows/main.yml:19]: Wouldn't be easier to just start `docker-compose up -d db cache search` and keep dependencies maintained in `docker-compose.yml` only?

**Pull request closed without merging**